pkg/auth/providers/goauth/goauth_provider.go

```diff
@@ -151,9 +146,12 @@ func (g *goauthProvider) SearchPrincipals(searchKey, principalType string, token
 		return nil, err
 	}
 	logrus.Debugf("[Google OAuth] SearchPrincipals: Retrieved stored oauth token")
-	if err = g.directoryServiceFromStoredToken(storedOauthToken, config); err != nil {
-		return principals, err
+	if g.adminSvc == nil {
+		if err = g.directoryServiceFromStoredToken(storedOauthToken, config); err != nil {
+			return principals, err
+		}
 	}
+
```

Comment:
Is the adminSvc instance initialised on every request? Is it supposed to be specific to each user?

Reply:
AdminSvc is checked/initialized during these requests:
Login: It has to be initialized and is specific to each user if only oauth creds are provided. If service account creds are provided, then it does not need to be initialized for every login as it's not specific to a user

Search/Get principals: It is initialized only if it's nil, which could happen if the in-memory struct storing it doesn't have it's value anymore (after server restart for instance).

Reply:
For Search/Get principals, so are we reusing the adminSvc between requests coming from different users? Say adminSvc is initialized by User1's token and saved on the provider struct, and if the next request for searchRequest is made by user2? Sounds like it should be initialized using the request's token.

Reply:
@prachidamle Changed this to init on every search/get call if oauth token is being used (not for svc creds)